src/components/nav/nav-controller.ts

```diff
@@ -1779,6 +1781,7 @@ export interface NavOptions {
   postLoad?: Function;
   progressAnimation?: boolean;
   climbNav?: boolean;
+  transitionData?: any;
```

Comment:
Whenever you `push` a page or `present` a view controller, you can pass `NavOptions`.  Ultimately, the Nav process ensures the property is set on the `TransitionOptions` object.